zap/src/output/luau/server.rs

```diff
@@ -532,6 +535,27 @@ impl<'a> ServerOutput<'a> {
 		self.push_line("incoming_read = 0");
 		self.push_line("incoming_ipos = 0");
 
+		if ev.evty == EvType::Unreliable(true) {
+			self.push_line("local saved = save()");
+			self.push_line("load_player(player)");
+			self.push_line(&format!(
+				"local order_id = buffer.read{UNRELIABLE_ORDER_NUMTY}(incoming_buff, read({}))",
+				UNRELIABLE_ORDER_NUMTY.size()
+			));
+			let last = format!("incoming_ids[{id}]");
+			self.push_line(&format!(
+				"if {last} and order_id <= {last} and {last} - order_id < {} then",
+				(UNRELIABLE_ORDER_NUMTY.max() / 2.0).floor(),
+			));
+			self.indent();
+			self.push_line("return");
+			self.dedent();
+			self.push_line("end");
+			self.push_line(&format!("{last} = order_id"));
+			self.push_line("player_map[player] = save()");
+			self.push_line("load(saved)");
```

Comment:
Calling save twice is hacky!
If you don't want to increment when calling `read`, just use the `incoming_read` variable.